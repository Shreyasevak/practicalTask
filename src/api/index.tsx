import axios, { } from 'axios';
import { studentList } from '../utils/types';

let baseUrl = 'https://reqres.in/api'

export const getStudentList = (page: number): Promise<studentList[]> => {
  return new Promise((resolve) => {
    axios.get(`${baseUrl}/users?page=${page}`).then(r => {
      resolve(r.data.data)
    }).catch((e) => { console.log('e', e) });
  })
};
