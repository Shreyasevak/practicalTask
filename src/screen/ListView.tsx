import React, { useEffect, useRef, useState } from 'react';
import {
    Dimensions,
    StyleSheet,
    View,
    FlatList,
    ActivityIndicator,
    SafeAreaView,
    StatusBar,
} from 'react-native';
import listItem from '../components/listItem';
import { getStudentList } from '../api';
import { studentList } from '../utils/types';
const { height } = Dimensions.get('window');

const ListView = () => {
    const [loader, setLoader] = useState<boolean>(false);
    const [studentData, setStudentData] = useState<studentList[]>([]);
    let pageRef = useRef<number>(1);
    useEffect(() => {
        setLoader(true);
        // API call
        getStudentList(pageRef.current).then(response => {
            setLoader(false);
            setStudentData(response); // Data store in state for the disaply
        });
    }, []);
    return (
        <>
        <StatusBar/>
        <SafeAreaView style={styles.center}>
            {!loader && studentData.length > 0 ? (
                <FlatList
                    showsVerticalScrollIndicator={false}
                    style={{ width: '100%',marginTop:10 }}
                    data={studentData}
                    renderItem={listItem}
                />
            ) : (
                <View
                    style={styles.loaderContainer}>
                    <ActivityIndicator size={20} color={'white'}></ActivityIndicator>
                </View>
            )}
        </SafeAreaView></>
    );
};

const styles = StyleSheet.create({
    center: {
        width: '100%',
        height: height,
        paddingHorizontal: 20,
    },
    loaderContainer: {
        width: '100%',
        height: '100%',
        alignItems: 'center',
        justifyContent: 'center',
    }
});

export default ListView;
