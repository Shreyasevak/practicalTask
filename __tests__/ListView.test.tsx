// __tests__/Demo.test.js
import 'react-native';
import React from 'react';
import { render, waitFor } from '@testing-library/react-native';
import ListView from '../src/screen/ListView'; // Replace with the correct path to your Demo component
import { getStudentList } from '../src/api';

// Mock the API function
jest.mock('../src/api', () => ({
    getStudentList: jest.fn(),
}));

test('renders list correctly when data is available', async () => {
    // Mock API response
    (getStudentList as jest.MockedFunction<typeof getStudentList>).mockResolvedValueOnce([
        {
            id: 1,
            email: 'george.bluth@reqres.in',
            first_name: 'George',
            last_name: 'Bluth',
            avatar: 'https://reqres.in/img/faces/1-image.jpg',
        },
        {
            id: 2,
            email: 'janet.weaver@reqres.in',
            first_name: 'Janet',
            last_name: 'Weaver',
            avatar: 'https://reqres.in/img/faces/2-image.jpg',
        },
    ]);
    const { toJSON } = render(<ListView />);


    // Render the component
    // Wait for asynchronous code to settle
    await waitFor(() => {
        // Assert that list items are present
        // expect(getByText('George')).toBeDefined();
        // expect(getByText('Janet')).toBeDefined();
        expect(toJSON).toMatchSnapshot()
    });

});
