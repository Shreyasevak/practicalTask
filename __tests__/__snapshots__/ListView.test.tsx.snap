// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders list correctly when data is available 1`] = `[Function]`;
